startform/src/main/kotlin/nl/nlportal/startform/service/StartFormService.kt

```diff
@@ -0,0 +1,118 @@
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.nlportal.startform.service
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.node.ObjectNode
+import jakarta.persistence.EntityNotFoundException
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import nl.nlportal.commonground.authentication.BedrijfAuthentication
+import nl.nlportal.commonground.authentication.BurgerAuthentication
+import nl.nlportal.commonground.authentication.CommonGroundAuthentication
+import nl.nlportal.commonground.authentication.exception.UserTypeUnsupportedException
+import nl.nlportal.startform.autoconfigure.StartFormConfig
+import nl.nlportal.startform.domain.StartForm
+import nl.nlportal.startform.domain.AanvragerIdentificatie
+import nl.nlportal.startform.domain.StartFormDTO
+import nl.nlportal.startform.domain.StartFormObject
+import nl.nlportal.startform.repository.StartFormRepository
+import nl.nlportal.zgw.objectenapi.domain.*
+import nl.nlportal.zgw.objectenapi.service.ObjectenApiService
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import java.util.UUID
+
+class StartFormService(
+    private val objectMapper: ObjectMapper,
+    private val objectenApiService: ObjectenApiService,
+    private val startFormConfig: StartFormConfig,
+    private val startFormRepository: StartFormRepository,
+) {
+    suspend fun saveFormToObjectsApi(
+        formName: String,
+        submissionData: ObjectNode,
+        authentication: CommonGroundAuthentication,
+    ): UUID {
+        val startForm = getStartFormByFormName(formName)
+
+        val startFormObject =
+            StartFormObject(
+                aanvragerIdentificatie(authentication),
+                submissionData,
+            )
+
+        val startFormRecord =
+            CreateObjectsApiObjectRequestRecordWithoutCorrection(
+                typeVersion = startForm.typeVersion,
+                data = objectMapper.valueToTree(startFormObject) as ObjectNode,
+                startAt = localDateTimeNowObejctsApiFormat(),
+            )
+
+        val objectsApiObjectRequest =
+            CreateObjectsApiObjectRequestWithoutCorrection(
+                uuid = UUID.randomUUID(),
+                type = "${startFormConfig.typeBaseUrl}/objecttypes/${startForm.typeUUID}",
+                record = startFormRecord,
+            )
+
+        val createdObject = objectenApiService.createObjectWithoutCorrection(objectsApiObjectRequest)
+        return createdObject.uuid
+    }
+
+    private fun localDateTimeNowObejctsApiFormat(): String {
+        val now = LocalDateTime.now()
+        val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
+        return now.format(formatter)
```

Comment:
Dit gaat denk ik niet geheel correct om 23:00/0:00 uur

Reply:
Volgens mij moet je gewoon LocalDate.now().toString() doen om de correcte datum formaat te krijgen, aangezien het default format yyyy-MM-dd is.
Op andere plaatsen in de library wordt ook gewoon LocalDate.now() gebruikt zonder extra formatting

Reply:
Helaas accepteerd de object api alleen het format wat hierboven te zien is